R/get_available_datasets.R

```diff
@@ -25,34 +32,41 @@
 #'
 #' # see only regional level datasets
 #' get_available_datasets("regional")
-get_available_datasets <- function(type) {
-  envi <- ls(getNamespace("covidregionaldata"), all.names = TRUE)
-  # regional data
-  starts_with_capitals_idx <- grep("^[A-Z]", envi)
-  starts_with_capitals <- envi[starts_with_capitals_idx]
-  exclude <- c("DataClass", "CountryDataClass")
-  valid_country_objects <- lapply(
-    starts_with_capitals,
-    function(x) {
-      country_obj <- get(x)
-      if (class(country_obj) == "R6ClassGenerator" & !(x %in% c(exclude))) {
-        dat <- get(x)$new()
-        dat <- dat$summary()
-        return(dat)
+#'
+#' # render the data
+#' get_available_datasets(render = TRUE)
+get_available_datasets <- function(type, render = FALSE,
+                                   namespace = "covidregionaldata") {
+  if (render) {
+    envi <- ls(getNamespace(namespace), all.names = TRUE)
+    starts_with_capitals_idx <- grep("^[A-Z]", envi)
+    starts_with_capitals <- envi[starts_with_capitals_idx]
+    exclude <- c("DataClass", "CountryDataClass")
+    valid_country_objects <- lapply(
+      starts_with_capitals,
+      function(x) {
+        country_obj <- get(x)
+        if (class(country_obj) == "R6ClassGenerator" & !(x %in% c(exclude))) {
+          dat <- get(x)$new()
+          dat <- dat$summary()
+          return(dat)
+        }
       }
-    }
-  )
-  available_country_data <- valid_country_objects %>%
-    bind_rows()
-
+    )
+    available_country_data <- valid_country_objects %>%
+      bind_rows()
+    country_data <- available_country_data
+  } else {
+    country_data <- all_country_data
```

Comment:
if we could possibly load the data usinng the covidregionaldata namespace. That may not be at all possible though so no sweat (I can't think how to do it certainly).

Reply:
So just replace with `covidregionaldata::all_country_data` ?

Reply:
So ideally specifying the `library` bit of this but I can't see how to do it so no worries for now.